src/databricks/labs/remorph/cli.py

```diff
@@ -107,70 +113,175 @@ def transpile(
 ):
     """Transpiles source dialect to databricks dialect"""
     ctx = ApplicationContext(w)
-    logger.debug(f"Application transpiler config: {ctx.transpile_config}")
-    checker = _TranspileConfigChecker(ctx.transpile_config)
-    checker.use_transpiler_config_path(transpiler_config_path)
-    checker.use_source_dialect(source_dialect)
-    checker.use_input_source(input_source)
-    checker.use_output_folder(output_folder)
-    checker.use_error_file_path(error_file_path)
-    checker.use_skip_validation(skip_validation)
-    checker.use_catalog_name(catalog_name)
-    checker.use_schema_name(schema_name)
+    print(f"{ctx.transpile_config!s}")
+    checker = _TranspileConfigChecker(ctx.transpile_config, ctx.prompts)
+    checker.check_input_source(input_source)
+    checker.check_source_dialect(source_dialect)
+    checker.check_transpiler_config_path(transpiler_config_path)
+    checker.check_transpiler_config_options()
+    checker.check_output_folder(output_folder)
+    checker.check_error_file_path(error_file_path)
+    checker.check_skip_validation(skip_validation)
+    checker.check_catalog_name(catalog_name)
+    checker.check_schema_name(schema_name)
     config, engine = checker.check()
     asyncio.run(_transpile(ctx, config, engine))
 
 
 class _TranspileConfigChecker:
```

Comment:
nit: I like this to moved inside a singleton Application Context!!! to keep the cli.py small.
non blocking